Spigot-Server-Patches/0594-Item-no-age-no-player-pickup.patch

```diff
@@ -0,0 +1,50 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Alfie Smith <alfie@alfiesmith.net>
+Date: Sat, 7 Nov 2020 01:20:33 +0000
+Subject: [PATCH] Item no age & no player pickup
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+index d6a64e4ecf9225f9a93523c50b2f1375296b406e..f34cfc5695c88754d7b78f49be282fb125a1cf94 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+@@ -10,6 +10,12 @@ import org.bukkit.entity.Item;
+ import org.bukkit.inventory.ItemStack;
+ 
+ public class CraftItem extends CraftEntity implements Item {
++
++    // Paper start
++    private final static int noAgeTime = -32768;
++    private final static int noPickupTime = 32767;
++    // Paper end
++
+     private final EntityItem item;
+ 
+     public CraftItem(CraftServer server, Entity entity, EntityItem item) {
+@@ -57,6 +63,26 @@ public class CraftItem extends CraftEntity implements Item {
+     public void setCanMobPickup(boolean canMobPickup) {
+         item.canMobPickup = canMobPickup;
+     }
++
++     @Override
++     public boolean canPlayerPickup() {
++        return item.pickupDelay == noPickupTime;
++     }
++
++     @Override
++     public void setCanPlayerPickup(boolean noPickup) {
++        item.pickupDelay = noPickup ? noPickupTime : 0;
++     }
++
++     @Override
++     public boolean isAgeing() {
++        return item.age == noAgeTime;
++     }
++
++     @Override
++     public void setAgeing(boolean ageing) {
++        item.age = ageing ? 0 : noAgeTime;
```

Comment:
Isn't this modified every tick to account for specifically ageing?

Reply:
I'm unsure what you mean by this. The age isn't changed if the value age is `noAgeTime`. It does increment every tick otherwise